### PR TITLE
adopt Anton's vcflib tools

### PR DESCRIPTION
**Pull request by martenson**

### CODE REVIEW

**Comment by jgoecks** [comment]
VCFfilter should accept a VCF dataset rather than VCF_bgzip dataset and the wrapper should compress and index using bgzip and tabix. Here's a quick diff that demonstrates this change: all that's missing is the tabix tool dependencies:

```python
diff -r 0e7ec249e679 vcffilter.xml
--- a/vcffilter.xml     Mon Sep 15 14:51:08 2014 -0400
+++ b/vcffilter.xml     Fri Mar 13 10:59:07 2015 -0400
@@ -5,8 +5,9 @@
   <description>filter VCF data in a variety of attributes</description>
   <command>
   <!-- This tools depends on tabix functionality, which is currently distributed with Galaxy itself via a pysam egg -->
-    ln -s "${input1}" input1.vcf.gz &amp;&amp;
-    ln -s "${Tabixized_input}" input1.vcf.gz.tbi &amp;&amp;
+    ln -s "${input1}" input1.vcf &amp;&amp;
+    bgzip input1.vcf &amp;&amp;
+    tabix -p vcf input1.vcf.gz &amp;&amp;
     vcffilter ${filterList} input1.vcf.gz  > "${out_file1}"
   </command>

@@ -21,8 +22,7 @@
         </mapping>
       </sanitizer>
     </param>
-    <param format="vcf_bgzip" name="input1" type="data" label="VCF dataset to filter">
-      <conversion name="Tabixized_input" type="tabix" />
+    <param format="vcf" name="input1" type="data" label="VCF dataset to filter">
     </param>
   </inputs>
   <outputs>
```

**Comment by martenson** [comment]
@jgoecks updated the PR with your diff and the needed dependency

**Comment by jgoecks** [comment]
Fantastic, thanks @martenson !

File: suites/suite_vcflib_14_08/tool_dependencies.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<repositories>
+    <repository name="vcf2tsv" owner="devteam" />
```
Comment (smithcr) [inline suites/suite_vcflib_14_08/tool_dependencies.xml]: this should be a repository_dependencies.xml file

File: suites/suite_vcflib_14_08/tool_dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <repository name="vcfaddinfo" owner="devteam" />
+    <repository name="vcfallelicprimitives" owner="devteam" />
+    <repository name="vcfannotate" owner="devteam" />
+    <repository name="vcfannotategenotypes" owner="devteam" />
```
Comment (bgruening) [inline suites/suite_vcflib_14_08/tool_dependencies.xml]: this should be a repository_dependencies.xml file

File: tool_collections/vcflib/vcfaddinfo/vcfaddinfo.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<tool id="vcfaddinfo" name="VCFaddinfo:" version="0.0.2">
+<requirements>
+    <requirement type="package" version="8a5602bf07">vcflib</requirement>
+</requirements>
+  <description>Adds info fields from the second dataset which are not present in the first dataset</description>
```
Comment (bgruening) [inline tool_collections/vcflib/vcfaddinfo/vcfaddinfo.xml]: I think a macro zourld be a good idea here, but we can also work on this after merging.

File: tool_collections/vcflib/vcfallelicprimitives/tool_dependencies.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<tool_dependency>
```
Comment (bgruening) [inline tool_collections/vcflib/vcfallelicprimitives/tool_dependencies.xml]: This file could also be a symlink.

File: tool_collections/vcflib/vcfallelicprimitives/vcfallelicprimitives.xml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                            Note that in many cases, such as multisample VCFs, these won't
+                            be valid post-decomposition.  For biallelic loci in single-sample
+                            VCFs, they should be usable with caution.
+    -g, --keep-geno         Maintain genotype-level annotations when decomposing.  Similar
+                            caution should be used for this as for --keep-info.
+
+----
+
+Vcfallelicprimitives is a part of VCFlib toolkit developed by Erik Garrison (https://github.com/ekg/vcflib).
+
```
Comment (bgruening) [inline tool_collections/vcflib/vcfallelicprimitives/vcfallelicprimitives.xml]: Can we cite these tools? Maybe also in a macro file?

Comment (martenson) [inline tool_collections/vcflib/vcfallelicprimitives/vcfallelicprimitives.xml]: I think they are not published :( I will double check with @nekrut

File: suites/suite_vcflib_14_08/tool_dependencies.xml (hunk 2 above)
Comment (martenson) [inline suites/suite_vcflib_14_08/tool_dependencies.xml]: good catch!

**Comment by martenson** [comment]
closing in favor of https://github.com/galaxyproject/tools-devteam/pull/78

**Pull request closed without merging**